samples/caf_sensor_manager/src/modules/workload_sim.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define MODULE workload_sim
+#include <caf/events/module_state_event.h>
+#include <caf/events/sensor_data_aggregator_event.h>
+
+#define BUSY_TIME CONFIG_WORKLOAD_TIME_PER_SAMPLE
+#define WORKLOAD_SIMULATED_THREAD_STACK_SIZE 800
+#define WORKLOAD_SIMULATED_THREAD_PRIORITY 1
+
+LOG_MODULE_REGISTER(MODULE);
+struct workload {
+	const struct sensor_data_aggregator_event *event;
+	struct k_work work;
+};
+
+static struct workload workload;
+
+static K_THREAD_STACK_DEFINE(workload_simulated_thread_stack,
+			     WORKLOAD_SIMULATED_THREAD_STACK_SIZE);
+struct k_work_q workload_simulated_work_q;
+
+void simulated_work_handler(struct k_work *work)
+{
+	struct workload *workload = CONTAINER_OF(work, struct workload, work);
+
+	/* Simulate processing of received data. */
+	k_busy_wait(BUSY_TIME * workload->event->sample_cnt);
+
+	struct sensor_data_aggregator_release_buffer_event *release_evt =
+		new_sensor_data_aggregator_release_buffer_event();
+
+	release_evt->buf = workload->event->buf;
+	release_evt->sensor_descr = workload->event->sensor_descr;
+	EVENT_SUBMIT(release_evt);
+}
+
+static bool init(void)
+{
+	k_work_queue_init(&workload_simulated_work_q);
+	k_work_init(&workload.work, simulated_work_handler);
+	k_work_queue_start(&workload_simulated_work_q,
+			   workload_simulated_thread_stack,
+			   K_THREAD_STACK_SIZEOF(workload_simulated_thread_stack),
+			   WORKLOAD_SIMULATED_THREAD_PRIORITY,
+			   NULL);
+	return false;
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_sensor_data_aggregator_event(eh)) {
+		const struct sensor_data_aggregator_event *event =
+			cast_sensor_data_aggregator_event(eh);
+
+		workload.event = event;
```

Comment:
You have to allocate `struct workload` on stack - here. You can't use single workload object, you would overwrite data if new `sensor_data_aggregator_event` comes during simulating workload for the previous one. Blocker.